src/main/java/org/openrewrite/kotlin/KotlinParser.java

```diff
@@ -473,25 +490,38 @@ public CompiledSource parse(List<Parser.Input> sources, Disposable disposable, E
         List<FirFile> rawFir = FirUtilsKt.buildFirFromKtFiles(firSession, ktFiles);
         Pair<ScopeSession, List<FirFile>> result = AnalyseKt.runResolution(firSession, rawFir);
         AnalyseKt.runCheckers(firSession, result.getFirst(), result.getSecond(), diagnosticsReporter);
-//        ModuleCompilerAnalyzedOutput analyzedOutput = new ModuleCompilerAnalyzedOutput(firSession, result.getFirst(), result.getSecond());
-//        FirResult firResult = new FirResult(singletonList(analyzedOutput));
-//
-//        Fir2IrExtensions extensions = Fir2IrExtensions.Default.INSTANCE;
-//        Fir2IrConfiguration irConfiguration = new Fir2IrConfiguration(
-//                languageVersionSettings,
-//                compilerConfiguration.getBoolean(JVMConfigurationKeys.LINK_VIA_SIGNATURES),
-//                compilerConfiguration.putIfAbsent(EVALUATED_CONST_TRACKER, EvaluatedConstTracker.Companion.create())
-//        );
-//        // TODO: add generation extensions
-//        List<IrGenerationExtension> irGenerationExtensions = new ArrayList<>();
-//        Fir2IrActualizedResult actualizedResult = convertToIrAndActualizeForJvm(firResult, extensions, irConfiguration, irGenerationExtensions, diagnosticsReporter);
+        ModuleCompilerAnalyzedOutput analyzedOutput = new ModuleCompilerAnalyzedOutput(firSession, result.getFirst(), result.getSecond());
+        FirResult firResult = new FirResult(singletonList(analyzedOutput));
+
+        Fir2IrExtensions extensions = Fir2IrExtensions.Default.INSTANCE;
+        Fir2IrConfiguration irConfiguration = new Fir2IrConfiguration(
+                languageVersionSettings,
+                compilerConfiguration.getBoolean(JVMConfigurationKeys.LINK_VIA_SIGNATURES),
+                compilerConfiguration.putIfAbsent(EVALUATED_CONST_TRACKER, EvaluatedConstTracker.Companion.create())
+        );
+
+        List<IrGenerationExtension> irGenerationExtensions = IrGenerationExtension.Companion.getInstances(projectEnvironment.getProject());
+        Fir2IrActualizedResult actualizedResult;
+        try {
+            actualizedResult = convertToIrAndActualizeForJvm(firResult, extensions, irConfiguration, irGenerationExtensions, diagnosticsReporter);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Unsure if this is the appropriate way to deal with the exception.

Reply:
I'll look into incremental compilation, which will help a lot.

I found that the `convertToIrAndActualizeForJvm` only throws an exception when types cannot be bound. Meaning either a compilation error exists in the project or a test was setup incorrectly.

Now, that we've updated the sources to prevent CRLF, we can add the analysis phase and likely surface compilation errors too